src/pp.rs

```diff
@@ -333,6 +340,65 @@ impl<'a> DirectiveProcessor<'a> {
         })
     }
 
+    fn parse_elif_directive(&mut self, directive_location: Location) -> Step<()> {
+        self.skipping = true;
+
+        // Do checks that the #elif block is well structured even if skipping.
+        if let Some(block) = self.blocks.last() {
+            if block.had_else {
+                return Err(StepExit::Error((
+                    PreprocessorError::ElifAfterElse,
+                    directive_location,
+                )));
+            }
+
+            // The condition isn't parsed if it doesn't need to (and doesn't produce errors).
+            if block.outer_skipped || block.had_valid_segment {
+                return self.consume_until_newline();
+            }
+        } else {
+            return Err(StepExit::Error((
+                PreprocessorError::ElifOutsideOfBlock,
+                directive_location,
+            )));
+        };
```

Comment:
In the case where you have `if let Some(x) = y ... else return Err(z)` you can instead use:
```
let x = y.ok_or(z)?
```

See also: https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or